### ISSUE TITLE
Auth error codes (at least) are locale-converted

### ISSUE THREAD

**Issue by mikehardy** [firebase/firebase-android-sdk#2593]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs in __the code in this repository__. 
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 4.1.3 (current stable)
  * Firebase Component: Auth (at least, may be others?) (Database, Firestore, Storage, Functions, etc)
  * Component version: BoM 26.8.0

#### Steps to reproduce:

Switch your device locale to Turkish
Use an auth.signInWithEmailAndPassword but give it an invalid email
Check the error code - it will have the Turkish i-with-no-dots characters

This is problematic because you need to match multiple cases for the "same" error code (that differs only in the 'i' character)

#### Relevant Code:

Relevant downstream issue: https://github.com/invertase/react-native-firebase/issues/3917

I *think* this is because the error code is being passed through a Locale sensitive method (like toLowerCase) while using the default locale instead of Locale.ROOT - I've had this problem before in Android code.

The messages might be localized but the codes should *always* be the same regardless of locale I think?

Not the most important thing but a quick grep of the auth code for Locale.getDefault() in context of error codes might turn it up quickly, and swapping that for Locale.ROOT might be one of the easier fixes ever, which is always satisfying.

Cheers

**Comment by malcolmdeck** [comment]
Thanks for the report. I've filed b/185793543 to track this internally

**Comment by mikehardy** [comment]
@malcolmdeck did you notice anything in your codebase that looked like this? I was enabling more lint rules in the react-native-firebase repo yesterday prepping to release the react-native bindings for the v28.x.x features here and I noticed that this might have been on our side? 

https://github.com/invertase/react-native-firebase/commit/bef04fc4e38dda3e30017f522c9dbe8f4035fdd3#diff-742544c6681aeb98a76076270ee96c33db2652bb2663b3a3f7c946b34be530aeL1888-R1889

I think this is probably closable here unless you've actually found something

Can always reopen if feedback from downstream users indicates it's still happening but I think I got it. Cheers!
